src/browser/rvm/runtime_initiated_topics/app_assets.ts

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2017 OpenFin Inc.
+
+Licensed under OpenFin Commercial License you may not use this file except in compliance with your Commercial License.
+Please contact OpenFin Inc. at sales@openfin.co to obtain a Commercial License.
+*/
+import {rvmMessageBus, AppAssetsGetList} from '../rvm_message_bus';
+import * as log from '../../log';
+import * as  _ from 'underscore';
+
+interface CallbackObj {
+    successCB: Function;
+    failureCB: Function;
+}
+
+interface PendingRequestObj {
+    [key: string]: {
+        [key: string] : Array<CallbackObj>
+    };
+}
+
+/**
+ * Module to handle fetching of app assets from RVM
+ *
+ **/
+class AppAssetsFetcher {
```

Comment:
Yeah I like a class here, to make that closure a bit more explicit. If it was from scratch idk if it would be